### ISSUE TITLE
Patch to fix segfault

### ISSUE THREAD

**Issue by doublex** [libssh2/libssh2#142]
``` diff
*** userauth.c-old      2016-10-26 13:45:53.710755576 +0200
--- userauth.c  2016-10-26 13:46:18.670550777 +0200
*************** sign_frommemory(LIBSSH2_SESSION *session
*** 723,729 ****
      if (privkeyobj->signv(session, sig, sig_len, 1, &datavec,
                            &hostkey_abstract)) {
          if (privkeyobj->dtor) {
!             privkeyobj->dtor(session, abstract);
          }
          return -1;
      }
--- 723,729 ----
      if (privkeyobj->signv(session, sig, sig_len, 1, &datavec,
                            &hostkey_abstract)) {
          if (privkeyobj->dtor) {
!             privkeyobj->dtor(session, &hostkey_abstract);
          }
          return -1;
      }
```

**Comment by alamaison** [comment]
Could you open a Pull Request with this patch?  It makes it hard to review like this and even harder to merge

**Comment by willco007** [comment]
Was fixed by commit 0dd6f96.
